tests/test_defects.py

```diff
@@ -0,0 +1,151 @@
+# This file is part of ci_cpp.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import os
+import numpy as np
+import unittest
+
+
+import lsst.afw.math as afwMath
+import lsst.daf.persistence as dafPersist
+import lsst.ip.isr as ipIsr
+import lsst.utils.tests
+
+from lsst.utils import getPackageDir
+
+
+class DefectTestCases(lsst.utils.tests.TestCase):
+
+    def setUp(self):
+        """Setup butler and generate an ISR processed exposure.
+
+        Notes
+        -----
+        DMTN-101 3.1:
+
+        Divide each resulting master frame by its median.
+        """
+        repoDir = os.path.join(getPackageDir('ci_cpp_gen2'), "DATA")
+        calibDir = os.path.join(getPackageDir('ci_cpp_gen2'), "DATA", "calibs")
+        butler = dafPersist.Butler(repoDir, calibRoot=calibDir)
+
+        self.config = ipIsr.IsrTaskConfig()
+        self.config.doSaturation = True
+        self.config.doSuspect = True
+        self.config.doSetBadRegions = True
+        self.config.doOverscan = True
+        self.config.doBias = True
+        self.config.doVariance = True
+        self.config.doDark = True
+        self.config.doFlat = True
+        self.config.doDefect = True
+
+        self.config.doLinearize = False
+        self.config.doCrosstalk = False
+        self.config.doWidenSaturationTrails = False
+        self.config.doBrighterFatter = False
+        self.config.doSaturationInterpolation = False
+        self.config.doStrayLight = False
+        self.config.doApplyGains = False
+        self.config.doFringe = False
+        self.config.doMeasureBackground = False
+        self.config.doVignette = False
+        self.config.doAttachTransmissionCurve = False
+        self.config.doUseOpticsTransmission = False
+        self.config.doUseFilterTransmission = False
+        self.config.doUseSensorTransmission = False
+        self.config.doUseAtmosphereTransmission = False
+
+        self.isrTask = ipIsr.IsrTask(config=self.config)
+        # This is not an independent frame.
```

Comment:
What does this mean? ("this is not an independent frame")

Reply:
This is one of the frames that went into the construction of the defects.  The majority of the test descriptions want the test performed on frames that were not used in the construction (difficult here due to data restrictions).  I'm not sure it matters here.

Reply:
Ah, I see my comments above came up here already. I stand by my first one though about maybe making this a ticket, which we can then consider.